plugins/source/aws/resources/services/s3/models/s3.go

```diff
@@ -29,6 +29,7 @@ type WrappedBucket struct {
 	LoggingTargetBucket   *string
 	LoggingTargetPrefix   *string
 	Policy                map[string]any
+	IsPublicByPolicy      bool
```

Comment:
Could we please change this to contain the entire `PolicyStatus` object, so it gets stored as a JSON column called `policy_status`?

I'm aware this may seems like an odd request. This will mean queries will need to be written as `policy_status->is_public` instead of `is_public_by_policy`, but it conforms better to our current standards, which are:
 - stick as closely as possible to the original AWS SDK method / CLI command names
 - store nested objects as JSON columns rather than unnesting them

Both these are in place so that it's easier to map from the AWS API to CloudQuery table columns.

The S3 table happens to be one of our oldest tables, so for backwards-compatibility reasons we haven't updated all columns to these standards yet, and what you've done here is totally reasonable when looking at the other columns in the table. But I think for setting the precedent, new columns we should try and follow the rules above, which would mean storing this additional information as a JSON column. One other advantage of this approach is also that, if AWS were ever to add new fields to that API response, they will automatically be included when we upgrade the AWS SDK version without requiring a manual update here.

Reply:
Thank you for review and I got it!

> if AWS were ever to add new fields to that API response, they will automatically be included when we upgrade the AWS SDK version without requiring a manual update here.

It means I should not use new struct, but use `map[string]any`, right?
so I'll change it as below. 
```go
  PolicyStatus map[string]any <- the result of `json.Unmarshal`
```

Reply:
@tsutsun17 I think you should be able to do:

```
PolicyStatus          *types.PolicyStatus
```

and set with:

```
resource.PolicyStatus = policyStatusOutput.PolicyStatus
```

Worth testing this though of course :)